apps/login/package.json

```diff
@@ -3,15 +3,26 @@
   "private": true,
   "scripts": {
     "dev": "next dev",
-    "test": "jest",
-    "test:watch": "jest --watch",
+    "test": "concurrently --group --timings --kill-others-on-fail 'npm:test:unit' 'npm:test:integration'",
+    "test:watch": "concurrently --kill-others 'npm:test:unit:watch' 'npm:test:integration:watch'",
+    "test:unit": "jest --config ./__test__/jest.config.ts",
+    "test:unit:watch": "pnpm test:unit --watch",
+    "test:integration": "export ZITADEL_API_URL=http://localhost:22222; pnpm mock:build && concurrently --group --names 'mock,test' --success command-test --kill-others 'pnpm:mock' 'start-server-and-test start http://localhost:3000 \"test:integration:run\"'",
+    "test:integration:watch": "concurrently --names 'mock,test' --kill-others 'pnpm:mock' 'ZITADEL_API_URL=http://localhost:22222 start-server-and-test dev http://localhost:3000 \"pnpm nodemon -e js,jsx,ts,tsx,css,scss --ignore \\\"__test__/**\\\" --exec \\\"pnpm test:integration:run\\\"\"'",
+    "test:integration:run": "cypress run --config-file ./cypress/cypress.config.ts --quiet",
+    "test:integration:open": "cypress open --config-file ./cypress/cypress.config.ts",
+    "mock": "pnpm mock:build && pnpm mock:run",
+    "mock:run": "pnpm mock:destroy && docker run --rm --name zitadel-mock-grpc-server --publish 22220:22220 --publish 22222:22222 zitadel-mock-grpc-server",
```

Comment:
I guess we dont need the 22220 port as we use 22222 right?

Reply:
22222 is the gRPC port.
22220 is the HTTP port for on-the-fly stubs administration.
We need both.

Reply:
ok got it 👍